course/enrollment.py

```diff
@@ -213,11 +270,12 @@ def enroll_view(request, course_identifier):
     if (
             preapproval is None
             and course.enrollment_required_email_suffix
-            and not user.email.endswith(course.enrollment_required_email_suffix)):
+            and not user.email.endswith(
+                "@%s" % course.enrollment_required_email_suffix.lstrip("@"))):
```

Comment:
This fix both `@abcd.com` `@bcd.com`  match `bcd.com` as suffix.

Reply:
And it eliminates the option of matching (say) `@math.somewhere.edu` and `@cs.somewhere.edu` with just `somewhere.edu`. Regular expressions would be the real solution here...

Reply:
I'm mainly skeptical because it's a change in behavior.

Reply:
What about this? We only need to match if it endswith ".somewhere.edu" or "@somewhere.edu"

Reply:
LGTM.